src/blob/core.rs

```diff
@@ -229,11 +235,16 @@ where
         // @TODO implement
     }
 
-    pub(crate) async fn write(&mut self, mut record: Record) -> Result<()> {
+    pub(crate) async fn write(blob: &ASRwLock<Self>, record: Record) -> Result<()> {
+        debug!("blob write");
+        let blob = blob.upgradable_read().await;
```

Comment:
Please, add comment that only one thread can acquire an `upgradable_read` lock. Because this is not obvious thing and the further code depends on this behavior (only one thread can write data at `self.current_offset`)